Spanner/src/Session/CacheSessionPool.php

```diff
@@ -718,7 +722,11 @@ private function isSessionValid(array $session)
     {
         $halfHourBeforeExpiration = $session['expiration'] - (SessionPoolInterface::SESSION_EXPIRATION_SECONDS / 2);
 
-        if ($this->time() < $halfHourBeforeExpiration) {
+        if (self::DURATION_SESSION_LIFETIME + $session['creation'] <
+            $this->time() + SessionPoolInterface::SESSION_EXPIRATION_SECONDS) {
+            // sessions more than 28 days old are auto deleted by server
+            return false;
+        } elseif ($this->time() < $halfHourBeforeExpiration) {
```

Comment:
This is a little confusing to me. Why is this check necessary? Shouldn't we only delete sessions older than 28 days once they've expired? This logic aggressively deletes sessions within `SESSION_EXPIRATION_SECONDS` before they are 28-days old, essentially making sure that they cannot ever reach more than 28 days old and still be valid. Is that WAD? 

It seems like this logic would be better like this:

```php
// session expires in more than a half hour
if ($this->time() < $halfHourBeforeExpiration) {
    return true;
}
// session expires in less than a half hour, but is not expired
if ($this->time() < $session['expiration']) {
   return $this->database
      ->session($session['name'])
      ->exists();
}
// no need to check if it's > 28 days old here, we do that later in the `maintain` function.
return false;
```

Note: I also cleaned up the conditional here because it's confusing - first, we should be using implicit `elseif` (meaning, if we return, we don't need an `elseif` because the function already returned).
Second, there's no need to check `$halfHourBeforeExpiration < $this->time()` when we already checked if `$this->time() < $halfHourBeforeExpiration` in the conditional beforehand.

Reply:
Yes we should aggressively delete 28d+ sessions. 
But it makes sense to not add a check here, since users will run `maintain` frequently.